torrent.go

```diff
@@ -1444,7 +1444,7 @@ func (t *Torrent) seeding() bool {
 	if cl.config.NoUpload {
 		return false
 	}
-	if !cl.config.Seed {
+	if !cl.IsSeeding() {
```

Comment:
I don't know if it's okay to call a method here that obtains a read lock on the Client.

Reply:
Why not ? Does it acquire recursive read lock here ? 

Oh, Can we have seperate mutex lock for Seeding ?

Reply:
@anacrolix  I think next commit solves it

Reply:
I don't think a separate mutex is the requirement, it's recursively locking it.

Reply:
@anacrolix oh,ok. what might be optimal solution for this?